control_panel_api/models.py

```diff
@@ -116,5 +146,23 @@ class Meta:
     def has_readwrite_access(self):
         return self.access_level == self.READWRITE
 
-    def update_aws_permissions(self):
-        services.apps3bucket_update(self)
+    def aws_create(self):
+        services.attach_bucket_access_to_app_role(
```

Comment:
(thinking aloud, **ignore me**): Wouldn't be nicer if these functions were called:
- `attach_bucket_access_to_app_role` => `grant_bucket_access_to_app`
- `detach_bucket_access_from_app_role` => `revoke_bucket_access_from_app`

In the future it may be nice to have an higher level class (`AWSAppS3Bucket`?), something like:

```python
AWSAppS3Bucket.from(apps3bucket).grant_access()
AWSAppS3Bucket.from(apps3bucket).update_access()
AWSAppS3Bucket.from(apps3bucket).revoke_access()

AWSS3Bucket.from(s3bucket).create()
AWSS3Bucket.from(s3bucket).delete()

AWSApp.from(app).create()
AWSApp.from(app).delete()
```

Reply:
I'm glad I have your complete approval on this @andyhd 🤣

Reply:
Shame `from` is a reserved keyword in Python 🤓, as I said, I was pretty much just thinking aloud. And I'm not suggesting to change this **now**.

At the same time, I still think there we're missing a trick here, some kind of entity which:
 - knows db models
 - knows how to update AWS to reflect the state of things in the CP-API DB

Whether this makes sense, how to call these things, how to call their methods I don't know yet.

Reply:
So right now we are at least HONEST.